### PR TITLE
cmake: Add options to support wolfTPM

### PR DESCRIPTION
**Pull request by elms**
Add options
 * certgen
 * certgencache
 * certreq
 * certext
 * cryptocb
 * pkcs7
 * X9.63 KDF
 * AES-CFB

zd 13379

# Testing

Tested with wolfTPM on windows using cmake with MSVC and Visual Studio. Also tested with wolfTPM on windows under msys.

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Review by dgarske — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -399,6 +399,8 @@ endif()
 #       - RIPEMD
 #       - BLAKE2
 
+set(WOLFSSL_AESCFB_HELP_STRING "Enable wolfSSL AES-CFB support (default: disabled)")
```
Comment (haydenroche5) [inline CMakeLists.txt:402]: Can you remove AES-CFB from the TODO above?

File: CMakeLists.txt [hunk 2]
```diff
@@ -488,6 +490,18 @@ endif()
 #       - Decoded cert cache
 #       - SEP
 
+set(WOLFSSL_CERTGEN_HELP_STRING "Enable cert generation (default: disabled)")
```
Comment (haydenroche5) [inline CMakeLists.txt:493]: Likewise for the TODOs here and elsewhere.

File: CMakeLists.txt [hunk 3]
```diff
@@ -1439,6 +1470,59 @@ add_option("WOLFSSL_USER_SETTINGS" ${WOLFSSL_USER_SETTINGS_HELP_STRING} "no" "ye
 set(WOLFSSL_OPTFLAGS_HELP_STRING "Enable default optimization CFLAGS for the compiler (default: enabled)")
 add_option("WOLFSSL_OPTFLAGS" ${WOLFSSL_OPTFLAGS_HELP_STRING} "yes" "yes;no")
 
+# FLAGS operations
+
+if(WOLFSSL_TPM)
+  override_cache(WOLFSSL_CERTGEN "yes")
```
Comment (haydenroche5) [inline CMakeLists.txt:1476]: 4 space indent here and elsewhere.

**Review by haydenroche5 — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (elms) [reply CMakeLists.txt:402]: Done

File: CMakeLists.txt (hunk 2 above)
Comment (elms) [reply CMakeLists.txt:489]: Done

File: CMakeLists.txt (hunk 3 above)
Comment (elms) [reply CMakeLists.txt]: Done

**Review by haydenroche5 — APPROVED** [review]

**Review by dgarske — APPROVED** [review]

**Pull request merged**